internal/service/ipam/api_range_resource_test.go

```diff
@@ -0,0 +1,501 @@
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccRangeExclusionRanges("10.0.0.20", "10.0.0.8", "10.0.0.16", "10.0.0.12"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRangeExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "exclusion_ranges.0.start", "10.0.0.12"),
+					resource.TestCheckResourceAttr(resourceName, "exclusion_ranges.0.end", "10.0.0.16"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccRangeExclusionRanges("10.0.0.20", "10.0.0.8", "10.0.0.16", "10.0.0.14"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRangeExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "exclusion_ranges.0.start", "10.0.0.14"),
+					resource.TestCheckResourceAttr(resourceName, "exclusion_ranges.0.end", "10.0.0.16"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccRangeResource_Name(t *testing.T) {
+	var resourceName = "bloxone_ipam_range.test_name"
+	var v ipam.IpamsvcRange
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccRangeName("10.0.0.20", "10.0.0.8", "range-test"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRangeExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "name", "range-test"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccRangeName("10.0.0.20", "10.0.0.8", "range-test-1"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRangeExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "name", "range-test-1"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccRangeResource_Space(t *testing.T) {
+	var resourceName = "bloxone_ipam_range.test_space"
+	var v ipam.IpamsvcRange
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccRangeSpace("10.0.0.20", "10.0.0.8", "bloxone_ipam_ip_space.one"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRangeExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttrPair(resourceName, "space", "bloxone_ipam_ip_space.one", "id"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccRangeSpace("10.0.0.20", "10.0.0.8", "bloxone_ipam_ip_space.one"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRangeExists(context.Background(), resourceName, &v),
```

Comment:
since a new resource is created when `space` is changed, you should also test if previous is destroyed

Reply:
Updted the test.